src/Repository/DefaultSluggableRepository.php

```diff
@@ -4,26 +4,28 @@
 
 namespace Knp\DoctrineBehaviors\Repository;
 
-use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Bundle\DoctrineBundle\Registry;
 use Knp\DoctrineBehaviors\Contract\Entity\SluggableInterface;
 
 final class DefaultSluggableRepository
 {
     /**
-     * @var EntityManagerInterface
+     * @var Registry
      */
-    private $entityManager;
+    private $registry;
 
-    public function __construct(EntityManagerInterface $entityManager)
+    public function __construct(Registry $registry)
     {
-        $this->entityManager = $entityManager;
+        $this->registry = $registry;
     }
 
     public function isSlugUniqueFor(SluggableInterface $sluggable, string $uniqueSlug): bool
     {
         $entityClass = get_class($sluggable);
 
-        $queryBuilder = $this->entityManager->createQueryBuilder()
+        $queryBuilder = $this->registry
+            ->getManagerForClass($entityClass)
```

Comment:
Better decopule this logic to standalone line, as it is easy to oversee.

```php
$sluggableRepository = $this->registry->getManagerForClass($entityClass);
```

Also, this change should be done probably or all subscribers, right?

Reply:
Just trying to get all test jobs ok before making any other modification :)

any idea on how to solve this one ? https://travis-ci.org/KnpLabs/DoctrineBehaviors/jobs/638486665

Do i really need to add a condition on the manager instance ?

Reply:
@var annotation above the variable should solve that